lib/game_logic.rb

```diff
@@ -0,0 +1,279 @@
+# frozen_string_literal: true
+
+# Class cell for grid
+class Cell
+  attr_accessor :value
+  def initialize(value = ' ')
+    @value = value
+  end
+end
+
+# Class for intializing the players
+class Player
+  attr_reader :color, :name
+  def initialize(name, color)
+    @name = name
+    @color = color
+  end
+end
+
+# Class for initializing the Board
+class Board
+  attr_reader :grid
+  def initialize(grid = empty_grid)
+    @grid = grid
+  end
+
+  def empty_grid
+    Array.new(3) { Array.new(3) { Cell.new } }
+  end
+
+  def get_cell(x, y)
+    grid[x][y]
+  end
+
+  def set_cell(x, y, value)
+    return false if get_cell(x, y).value != ' '
+
+    get_cell(x, y).value = value
+    true
+  end
+
+  def game_over
+    return :winner if winner?
+
+    false
+  end
+
+  private
+
+  def winner?
+    for n in 0..2
+      return true if (get_cell(n, 0).value == get_cell(n, 1).value && get_cell(n, 1).value == get_cell(n, 2).value) && get_cell(n, 0).value != ' '
+    end
+    for n in 0..2
+      return true if (get_cell(0, n).value == get_cell(1, n).value && get_cell(1, n).value == get_cell(2, n).value) && get_cell(0, n).value != ' '
+
+    end
+    return true if ((get_cell(0, 0).value == get_cell(1, 1).value && get_cell(1, 1).value == get_cell(2, 2).value) && get_cell(0, 0).value != ' ') || ((get_cell(0, 2).value == get_cell(1, 1).value && get_cell(1, 1).value == get_cell(2, 0).value) && get_cell(1, 1).value != ' ')
+
+    false
+  end
+end
+
+# Class to initialize and control the game
+class Game
+  attr_reader :players, :board, :player_one, :player_two
+  def initialize(players = players_generator, board = Board.new)
+    @players = players
+    @board = board
+    @player_one, @player_two = players.shuffle
+  end
+
+  def turns_switch
+    @player_one, @player_two = @player_two, @player_one
+  end
+
+  def ask_move
+    [
+      "#{@player_one.name}, please enter a number from 1 to 9 to choose your position.",
+      "#{@player_one.name}, please select your next spot with a number between 1 and 9.",
+      "#{@player_one.name}, its you're turn! Choose a number between 1 and 9 to place your symbol.",
+      "#{@player_one.name}, time to pick a spot! Select a number 1 to 9 to choose your next move.",
+      "#{@player_one.name}, pick dat dere spot on dat dere board, wit dem numbers 'tween 1 and 9.",
+      "#{@player_one.name}, there are 9 spots. Many of these spots have been taken, the world waits, for you, at this very moment,
+      to make the ultimate decision. The ultimate sacrifice. Right here. Right now.\n\n Please enter a number from 1 to 9 to choose your position.",
+      "#{@player_one.name}, life is like a box of chocolates. Please enter a number from 1 to 9 to choose your position."
+    ]
+  end
```

Comment:
- No `puts/print` and `get.chomp` outside `main.rb` file. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/only_main_print.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/)). So move that part of code to the `main.rb`